### DISCUSSION TITLE
Token refresh

### DISCUSSION THREAD

**Discussion by tlemmons** [tsightler/ring-mqtt#432] [Support Q&A]
I put this in the other discussion but since it was closed I was not sure it would be seen.   

I have to regenerate my ring token often. Sometimes it lasts for a few days, sometimes a few minutes. I have tried to find the json config file on my system (virtualbox vm) and cannot find it anywhere on the vm file system. Not sure where to look. I have reinstalled the plugin a couple of times and nothing seems to stop the eventual failure. I am willing to do anything I can to help.

**Comment by tsightler** [comment]
Can you please describe in more detail your setup, i.e. the hardware, OS version, install method, etc.  I mean, you say you cannot find the ring-state.json file on your system, but it should be easy to find in all case since, for standard install, it's simply stored in the same folder as the ring-mqtt config file itself, for Docker install, it's stored in the persistent volume location that you have to specify, and for the addon it's stored in the standard /data folder.

Also, if you restart does it come back or do you always have to regenerate the token?  If you want you can send logs to my username at gmail.

**Comment by tlemmons** [comment]
Sorry, I should have done better.  I will send you some stuff asap.  I have looked at the entire vm filesystem and I don't find any ring-mqtt files.  Or any other addons.   So I figure I am looking at it wrong.  I do not intentionally have docker, just the virtualbox download from HA install website.  One thing I have not found is any good information about how the install is arraigned and where to find file.

**Reply by tsightler** [reply]
OK, so you're using Home Assistant OS running in VirtualBox so you are using the addon.  In HAOS the addons all run inside their own Docker container so you'll need to install the SSH addon and configure it to allow running docker commands, then you can access the ring-mqtt container to find the file.

One thing, are you sure that you have time sync configured correctly for the VirtualBox system?  Having proper time settings can be important for proper token management.
